src/components/SeniorView.tsx

```diff
@@ -26,9 +29,22 @@ export const SeniorView = ({ seniors, students }: SeniorViewProps) => {
 
   const [yearsClicked, setYearsClicked] = useState<number[]>([]);
 
+  const deleteSeniorIdRef = React.useRef("");
```

Comment:
Without ref, the callback stores an empty string.

Reply:
Should we consider passing back `seniorId` from backend? It feels quite janky to use ref here.